src/main/java/com/example/auctionapp/repository/BaseRepository.java

```diff
@@ -22,6 +22,14 @@
     @PersistenceContext
     EntityManager entityManager;
 
+    public BaseRepository() {
+    }
+
+    public BaseRepository(Class<T> resourceClass, EntityManager entityManager) {
+        this.resourceClass = resourceClass;
+        this.entityManager = entityManager;
+    }
+
     public void setResourceClass(Class< T > classToSet ) {
         this.resourceClass = classToSet;
     }
```

Comment:
We need to set up a convention regarding these classes.
Either use a `constructor`, or use an `empty constructor + setter`.
Now, as we want class instances that are not subject to change - their properties should be set once, we need to use constructors.
So please align all these classes - services, repositories to use a constructor only approach.

Reply:
That means removing setters, removing annotations from properties and introducing constructors with the same annotations.